### PR TITLE
fixed moment.js deprecated zone and add functions

### PR DESCRIPTION
**Pull request by Fish2**
#### Database Migration
NO

#### Description
fixed moment.js deprecated zone and add functions.

Had a lot of issues with the last one so made a fresh start and it is fully tested and working now.

#### Todos
- [x] Tests

#### Issues Fixed or Closed by this PR

* https://github.com/Sonarr/Sonarr/pull/2258

Related 
* https://github.com/Radarr/Radarr/pull/2232
* https://github.com/Radarr/Radarr/issues/2231

### CODE REVIEW

**Comment by Fish2** [comment]
Any news on pulling this in ?

File: src/UI/JsLibraries/fullcalendar.js [hunk 1]
```diff
@@ -914,12 +914,7 @@ function makeMoment(args, parseAsUTC, parseZone) {
 				mom._ambigZone = true;
 			}
 			else if (isSingleString) {
-				if (mom.utcOffset) {
```
Comment (markus101) [inline src/UI/JsLibraries/fullcalendar.js]: JS Libraries should not be manipulated, please revert this change.

**Comment by Fish2** [comment]
@markus101 no worries. 

Ok i will have to look into that as it was removed because it was causing issues i will get back to you in a hour or so

File: src/UI/JsLibraries/fullcalendar.js (hunk 1 above)
Comment (markus101) [reply src/UI/JsLibraries/fullcalendar.js]: This was not reverted properly, I've cleaned it and merged it in though.

**Comment by markus101** [comment]
Cleaned up and merged manually.
